operations/agent_monitor.go

```diff
@@ -276,6 +277,20 @@ func handleMonitorSignals(ctx context.Context, serviceCancel context.CancelFunc)
 func (m *monitor) fetchClient(ctx context.Context, urls []string, retry utility.RetryOptions) error {
 	var downloaded bool
 	catcher := grip.NewBasicCatcher()
+
+	// macOS caches signatures, which means that a newly downloaded
+	// evergreen agent can fail to start. Removing the agent before
+	// downloading prevents this from occurring. See
+	// https://apple.stackexchange.com/questions/258623/how-to-fix-killed-9-error-in-mac-os/428388#428388
+	if runtime.GOOS == "darwin" {
+		_, err := os.Stat(m.clientPath)
```

Comment:
Instead of this stat -> remove, can we just use `os.RemoveAll`? It already squashes the error when the file does not exist.